File: dotnet/Sers/Sers.Core/Sers.Core.Station/Module/SsApiDiscovery/LocalApiNode.cs
using System;
using System.Reflection;
using Newtonsoft.Json;
using Sers.Core.Module.Api.ApiDesc;
using Sers.Core.Module.Api.LocalApi;

namespace Sers.Core.Module.SsApiDiscovery
{
    [JsonObject(MemberSerialization.OptIn)]
    public class LocalApiNode: IApiNode
    {
        [JsonProperty]
        public SsApiDesc apiDesc { get; set; }

        public LocalApiNode(SsApiDesc apiDesc,  MethodInfo apiController_Method,Object apiController_Obj)
        {
            this.apiDesc = apiDesc;
            this.apiController_Method = apiController_Method;
            this.apiController_Obj = apiController_Obj;
        }

        public virtual byte[] Invoke(ArraySegment<byte> arg_OriData)
        {

            //(x.1)反序列化 请求参数
            var args = apiDesc.argType?.Deserialize(arg_OriData);

            //(x.2) Invoke
            var returnValue = apiController_Method.Invoke(apiController_Obj, args);
 
            //(x.3) 序列化 返回数据
            return Serialization.Serialization.Instance.Serialize(returnValue);
        }
                          

        #region apiController        

        //private Type apiController_Type;
        MethodInfo apiController_Method;
        Object apiController_Obj;
        #endregion


    }
}

File: java/Sers.ServiceStation/src/Sers/Core/Module/SsApiDiscovery/LocalApiNode.java
package Sers.Core.Module.SsApiDiscovery;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;

import Sers.Core.Module.Api.ApiDesc.SsApiDesc;
import Sers.Core.Module.Serialization.Serialization;
import Sers.Core.Util.Data.ArraySegment;

public class LocalApiNode {	

	private Method method;
	private IApiController controller;

	public SsApiDesc apiDesc;
	// public String route;
	
	public void  init(Method method,IApiController controller) {
		this.method=method;
		this.controller=controller;		
	}
	

	
	public byte[] invoke(ArraySegment arg_OriData) throws  Exception {
		
		 //(x.1)反序列化 请求参数
        Object[] args = apiDesc.argType.deserialize(arg_OriData);

        //(x.2) Invoke
        Object returnValue=method.invoke(controller,args);

        //(x.3) 序列化 返回数据
        return Serialization.Instance.serializeToBytes(returnValue);   
    
	}

}
